crates/editor/src/element.rs

```diff
@@ -1482,6 +1482,46 @@ impl EditorElement {
         axis_pair(horizontal_scrollbar, vertical_scrollbar)
     }
 
+    fn layout_minimap(
+        &self,
+        window: &mut Window,
+        cx: &mut App,
+        snapshot: &EditorSnapshot,
+        bounds: Bounds<Pixels>,
+        scrollbar_width: Pixels,
+    ) -> Option<AnyElement> {
+        match snapshot.mode {
+            EditorMode::Full => {
+                let mut editor = self
+                    .editor
+                    .update(cx, |editor, cx| editor.clone(window, cx));
+
+                editor.mode = EditorMode::Minimap;
+                editor.set_text_style_refinement(TextStyleRefinement {
+                    font_size: Some(px(2.).into()),
+                    ..Default::default()
+                });
+
+                let editor_entity = cx.new(|_| editor);
+
+                let mut minimap_elem = editor_entity.update(cx, |editor, cx| {
+                    editor.render(window, cx).into_any_element()
+                });
+
+                let mut minimap_bounds = bounds.clone();
+                minimap_bounds.size.width = px(100.);
+                minimap_bounds.origin.x =
+                    bounds.size.width - minimap_bounds.size.width - scrollbar_width;
+                _ = minimap_elem.layout_as_root(minimap_bounds.size.into(), window, cx);
```

Comment:
Interesting to note, ST2 and VSCode editors' minimaps do not scroll horizontally, while the current version does.
We might consider doing it too, as it simplified clicks on map impls.

Reply:
I made an attempt to correct this, however if you scroll far enough I think the buffer shifts anyway. I'm looking into this

Reply:
fixed, i disabled autoscroll